src/mapgen/dungeongen.cpp

```diff
@@ -51,7 +51,7 @@ DungeonGen::DungeonGen(const NodeDefManager *ndef,
 #endif
 
 	if (dparams) {
-		memcpy(&dp, dparams, sizeof(dp));
+		dp = *dparams;
```

Comment:
memcpy is necessary here. which warning do you want to fix ?

Reply:
the warning is that there is a non trivial copy constructor for this class